snakemake/workflow.py

```diff
@@ -1114,6 +1140,25 @@ def log_provenance_info():
                 if stats:
                     self.scheduler.stats.to_json(stats)
                 logger.logfile_hint()
+                if self.benchmark_all is not None or self.benchmark_output is not None:
```

Comment:
Benchmarks are currently only written `if success` is true. Would be good to write in any case, I think. While some rules might have failed, it might only be a last rule and benchmark might still be valuable.

Reply:
Updated to gather benchmarks whether success or not